path_login.go

```diff
@@ -176,6 +180,7 @@ func (b *azureAuthBackend) pathLogin(ctx context.Context, req *logical.Request,
 			Metadata: map[string]string{
 				"resource_group_name": resourceGroupName,
 				"subscription_id":     subscriptionID,
+				"app_id":              appID,
```

Comment:
Curious if there is any particular reason to put the app ID into the alias and token metadata?

Reply:
I believe the idea was that it could be part of the authorization decision now, so put it in there. I don't think we _have_ to have it?

I did refactor it a bit so that it should only set if present in the claim.

Reply:
We don't have to include it, but I like the idea of including it because it's a part of the authorization decision. I'd say let's leave it. Good idea to conditionally add it.